connectors/sources/github.py

```diff
@@ -714,15 +760,22 @@ async def graphql(self, query, variables=None):
         Yields:
             dictionary: Client response
         """
-        url = f"{self.github_url}/graphql"
+        url = f"{self.base_url}/graphql"
         self._logger.debug(
             f"Sending POST to {url} with query: '{json.dumps(query)}' and variables: '{json.dumps(variables)}'"
         )
         try:
+            self._get_client.oauth_token = self._access_token()
```

Comment:
Most of the methods in [gidgethub](https://github.com/gidgethub/gidgethub/blob/main/gidgethub/abc.py) accept params like `oauth_token` to override the default one, e.g. [getitem](https://github.com/gidgethub/gidgethub/blob/7f8cec45922056a38c51ece276844a4bc028e6f5/gidgethub/abc.py#L129-L138), but [graphql](https://github.com/gidgethub/gidgethub/blob/7f8cec45922056a38c51ece276844a4bc028e6f5/gidgethub/abc.py#L306-L312) doesn't and it [uses the default one](https://github.com/gidgethub/gidgethub/blob/7f8cec45922056a38c51ece276844a4bc028e6f5/gidgethub/abc.py#L323)

I update the default `oauth_token` every time we make a `graphql` call, but in the future, we can create a pull request in gidgethub

Reply:
I think it's also nice to just wrap all this logic (client calls + auth) into a class to make it easier to understand. Not for this PR probably, but for future it would help IMO

Reply:
Could you elaborate more?

The Client, `GitHubAPI` does accept `oauth_token` as parameter in `__init__`, as the default token, see [here](https://github.com/gidgethub/gidgethub/blob/7f8cec45922056a38c51ece276844a4bc028e6f5/gidgethub/abc.py#L44). And it does accept `oauth_token` in most methods, to override the default one, e.g. [getitem](https://github.com/gidgethub/gidgethub/blob/7f8cec45922056a38c51ece276844a4bc028e6f5/gidgethub/abc.py#L136), [getiter](https://github.com/gidgethub/gidgethub/blob/7f8cec45922056a38c51ece276844a4bc028e6f5/gidgethub/abc.py#L180), but not [graphql](https://github.com/gidgethub/gidgethub/blob/7f8cec45922056a38c51ece276844a4bc028e6f5/gidgethub/abc.py#L306)

I think the better approach is to file a PR in `gidgethub`, but I can't wait for it to be merged.

Reply:
I meant to make a client similar to what we have in [SPO](https://github.com/elastic/connectors/blob/main/connectors/sources/sharepoint_online.py#L521) - under the hood it tries to handle all inconsistencies and provide you with simple interface (doing token refreshes, choosing proper API under the hood).

It's of course possible to make a PR in 3rd-party lib but it definitely will not be fast